### ISSUE TITLE
Rename function argument scans entire workspace

### ISSUE THREAD

**Issue by zbs** [microsoft/pylance-release#882]
I find that most rename refactors are unreliable because the language server parses the entire workspace contents first. For example, for the given code

```
def foo(bar, baz):
  bar.append("something")
  baz.append("something else")
```
and for the following steps:

1. Put cursor in `bar` in function signature
2. Hit F2
3. Rename `bar` to `random`
4. Hit enter

VS Code hangs indefinitely. Looking in the `Language Server` section of output, it shows a log containing a long list of files that are being parsed/binded. If there's no way to automatically limit the scope of a refactor when appropriate, would it be possible to add a setting that constrains the refactor to the current file? Barring that, some sort of refactor preview, where you could select the affected files, would improve the refactor experience for me considerably. I assume the latter is outside the scope of Pylance.

**Comment by heejaechang** [comment]
we can do some text based filtering to reduce parsing (probably won't help much for simple name like "a" or "i") but otherwise, we need to parse a file to build dependency graph for externally visible symbols such as method parameter.

I will at least add text based filtering for now.

**Comment by heejaechang** [comment]
linking issue that has same root cause - https://github.com/microsoft/pylance-release/issues/2457

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.3.2, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202232-16-March-2022)
